tests/adapters/test_epicsadapter/test_epics_adapter.py

```diff
@@ -139,3 +152,31 @@ def test_epics_adapter_load_records_without_DTYP_fields_method(
         written_data = file.read()
 
     assert str(written_data).strip() == str(test_params["expected"]).strip()
+
+
+@pytest.mark.asyncio
+async def test_db_file_not_specified(
+    epics_adapter_no_db_file: EpicsAdapter,
+    MockDevice: Device,
+    mock_raise_interrupt: RaiseInterrupt,
+):
+    epics_adapter_no_db_file.on_db_load = MagicMock()
+    epics_adapter_no_db_file.load_records_without_DTYP_fields = MagicMock()
+
+    await epics_adapter_no_db_file.run_forever(MockDevice, mock_raise_interrupt)
+
+    epics_adapter_no_db_file.load_records_without_DTYP_fields.assert_not_called()
+
+
+@pytest.mark.asyncio
+async def test_db_load_called_if_file_specified(
```

Comment:
I'm not entirely happy with some of the duplication, is it worth somehow trying to parameterise, or extracting the common code to a separate method?

Reply:
I am not too worried about duplicated lines in tests. Where there is much of it it can be helpful to do what you suggest but for speed and clarity I'm happy enough with this.